tests/integration/targets/s3_sync/tasks/main.yml

```diff
@@ -55,6 +55,30 @@
           - output is changed
 
     # ============================================================
+    - name: Create a second s3_bucket
+      s3_bucket:
+        name: "{{ resource_prefix }}-testbucket-ansible-2"
+        state: present
+      register: output
+
+    - assert:
+        that:
+          - output.changed
+          - output.name == '{{ resource_prefix }}-testbucket-ansible-2'
+          - not output.requester_pays
+    
+    - name: Sync files with remote bucket using glacier storage class
+      s3_sync:
+        bucket: "{{ resource_prefix }}-testbucket-ansible-2"
+        file_root: "{{ output_dir }}/s3_sync"
+        storage_class: "GLACIER"
+      register: output
+
+    - assert:
+        that:
+          - output is changed
```

Comment:
Any way we can test that this is actually pushing to glacier?

Reply:
@tremble I verified in the AWS console. According to the documentation https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Bucket.upload_fileobj, this information doesn't seem to be returned, for this reason I didn't implement a check.